lib/kvdb/ikvdb.c

```diff
@@ -414,6 +414,77 @@ ikvdb_create(const char *kvdb_home, struct kvdb_cparams *params, bool pmem_only)
     return err;
 }
 
+merr_t
+ikvdb_attach(
+    const char *kvdb_home_tgt,
+    const char *kvdb_home_src,
+    const char *paths[HSE_MCLASS_COUNT])
+{
+    struct kvdb_meta    meta_tgt;
+    struct kvdb_meta    meta_src;
+    struct kvdb_rparams rp;
+    struct ikvdb       *kvdb;
+    merr_t err;
+    int    i;
+
+    INVARIANT(kvdb_home_tgt && kvdb_home_src && paths);
+
+    err = kvdb_meta_deserialize(&meta_src, kvdb_home_src);
+    if (err) {
+        log_errx("cannot attach KVDB (%s) from KVDB (%s), deserializing source meta failed: @@e",
+                 err, kvdb_home_tgt, kvdb_home_src);
+        return err;
+    }
+
+    /* Copy relevant fields from the src KVDB home meta file */
+    meta_tgt.km_version = meta_src.km_version;
+    meta_tgt.km_omf_version = meta_src.km_omf_version;
+    meta_tgt.km_cndb = meta_src.km_cndb;
+    meta_tgt.km_wal = meta_src.km_wal;
+
+    for (i = HSE_MCLASS_BASE; i < HSE_MCLASS_COUNT; i++) {
+        if ((meta_src.km_storage[i].path[0] == '\0' && paths[i] && paths[i][0] != '\0') ||
+            (meta_src.km_storage[i].path[0] != '\0' && (!paths[i] || paths[i][0] == '\0'))) {
+            log_err("cannot attach KVDB (%s) from KVDB (%s): %s", kvdb_home_tgt, kvdb_home_src,
```

Comment:
Start log message with a capital letter.

Reply:
I think log_err() prefixes a message with its "<func_name>:". In that case, it should be fine to start a log message with lowercase, right?

Reply:
Don't know off the top of my head, but we are generally inconsistent across the codebase.

Reply:
I generally don't care about captilalization of log messages, they are usually terse and not really proper sentences anyway..
Yes, log messages always include the function name.